sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -864,30 +865,16 @@ object RapidsBufferCatalog extends Logging {
   }
 
   private def closeImpl(): Unit = synchronized {
-    if (_singleton != null) {
-      _singleton.close()
-      _singleton = null
-    }
-
-    if (memoryEventHandler != null) {
-      // Workaround for shutdown ordering problems where device buffers allocated with this handler
-      // are being freed after the handler is destroyed
-      //Rmm.clearEventHandler()
-      memoryEventHandler = null
-    }
+    val memoryEventHandlerCloser: AutoCloseable = () =>
+      if (memoryEventHandler != null) {
+        // Workaround for shutdown ordering problems where device buffers allocated
+        // with this handler are being freed after the handler is destroyed
+        //Rmm.clearEventHandler()
+        memoryEventHandler = null
+      }
 
-    if (deviceStorage != null) {
-      deviceStorage.close()
-      deviceStorage = null
-    }
-    if (hostStorage != null) {
-      hostStorage.close()
-      hostStorage = null
-    }
-    if (diskStorage != null) {
-      diskStorage.close()
-      diskStorage = null
-    }
+    Seq(_singleton, memoryEventHandlerCloser, deviceStorage, hostStorage, diskStorage)
+      .filter(e => e != null).safeClose()
```

Comment:
I don't think the filter is necessary since safeClose handles null arguments.

Reply:
Thanks, removed it.